pkg/reconciler/domainmapping/reconciler.go

```diff
@@ -292,12 +297,15 @@ func (r *Reconciler) resolveRef(ctx context.Context, dm *v1alpha1.DomainMapping)
 
 func (r *Reconciler) reconcileDomainClaim(ctx context.Context, dm *v1alpha1.DomainMapping) error {
 	dc, err := r.netclient.NetworkingV1alpha1().ClusterDomainClaims().Get(ctx, dm.Name, metav1.GetOptions{})
+	if err != nil && !apierrs.IsNotFound(err) {
+		return fmt.Errorf("failed to get ClusterDomainClaim: %w", err)
+	}
+
 	if apierrs.IsNotFound(err) {
-		if dc, err = r.netclient.NetworkingV1alpha1().ClusterDomainClaims().Create(ctx, resources.MakeDomainClaim(dm), metav1.CreateOptions{}); err != nil {
-			return fmt.Errorf("failed to create ClusterDomainClaim: %w", err)
+		dc, err = r.createDomainClaimIfAllowed(ctx, dm)
+		if err != nil {
```

Comment:
fold assignment into `if`?

Reply:
Did it like this to try to make clear as possible that this is updating the outer dc and doesn't shadow it, but I'll happily change if you don't think this made that clearer

Reply:
I see, thanks. I don't mind keeping it this way, please feel free to do so.

But I think @vagababov will one day change it anyway 🤣.

Reply:
😂